### PR TITLE
Selection-playlist-test

### PR DESCRIPTION
**Pull request by RenBabiuch**

### CODE REVIEW

File: tests/PageObjects/SongListPage.ts [hunk 1]
```diff
@@ -108,12 +112,36 @@ export class SongListPagePO {
...
     await this.page.waitForTimeout(500);
     await this.page.keyboard.press('Backspace');
   }
+
+  public get selectionPlaylistTip() {
+    return this.page.locator("[role='tooltip']");
+  }
+
+  public async closeTheSelectionPlaylistTip() {
+    await this.selectionPlaylistTip.getByText('Close').click();
```
Comment (Asvarox) [inline tests/PageObjects/SongListPage.ts:125]: can you add the `data-test` attribute to the button>

File: tests/PageObjects/SongListPage.ts [hunk 2]
```diff
@@ -108,12 +112,36 @@ export class SongListPagePO {
...
+  }
+
+  public async expectSongToBeMarkedAsPopular(songID: string) {
+    await expect(this.getSongElement(songID).getByTestId('StarIcon')).toBeVisible();
```
Comment (Asvarox) [inline tests/PageObjects/SongListPage.ts:129]: `getByTestId` looks for `data-test` attribute and icons have `data-testid`

File: tests/selection-playlist.spec.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+const convertedSong = {
+  ID: 'selection-test-new-convert-song',
+  artist: 'Selection test',
+  title: 'New convert song',
+  author: 'Selection txt',
+  sourceURL: 'https://example.com/source-url',
+  videoID: 'W9nZ6u15yis',
+};
+
+const unknownSong = {
```
Comment (Asvarox) [inline tests/selection-playlist.spec.ts:36]: better call it e.g. `unpopularSong`

File: tests/selection-playlist.spec.ts [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+  await test.step('Search the song - expect song is not visible in Selection playlist yet', async () => {
+    await pages.songListPage.expectPlaylistToBeSelected(selectionPlaylist);
+    await pages.songListPage.closeTheSelectionPlaylistTip();
+    await expect(pages.songListPage.getSongElement(unknownSong.ID)).not.toBeVisible();
+    await pages.songListPage.searchSong(unknownSong.title);
+  });
+
+  await test.step('Toggle game mode to `Cooperation` and play the song', async () => {
+    await pages.songListPage.openPreviewForSong(unknownSong.ID);
+    await pages.songPreviewPage.toggleGameMode();
```
Comment (Asvarox) [inline tests/selection-playlist.spec.ts:153]: is this needed for the test?

File: tests/PageObjects/SongListPage.ts (hunk 1 above)
Comment (RenBabiuch) [reply tests/PageObjects/SongListPage.ts:125]: ok, done :)

File: tests/selection-playlist.spec.ts [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+  });
+
+  await test.step('Skip to the Song list', async () => {
+    await pages.postGameResultsPage.skipScoresAnimation();
+    await pages.postGameResultsPage.goToHighScoresStep();
+    await pages.postGameHighScoresPage.goToSongList();
+  });
+
+  await test.step('A song that is less then 80% complete, should not be added to the Selection playlist', async () => {});
+  await pages.songListPage.goToPlaylist(selectionPlaylist);
```
Comment (Asvarox) [inline tests/selection-playlist.spec.ts:172]: these instructions are outside the step (the step callback function is empty `async () => {}`)

File: tests/selection-playlist.spec.ts (hunk 4 above)
Comment (RenBabiuch) [reply tests/selection-playlist.spec.ts:153]: In this test, I am waiting for some amount of points and I have already prepared a method for it (using cooperation score)

File: tests/selection-playlist.spec.ts (hunk 5 above)
Comment (RenBabiuch) [reply tests/selection-playlist.spec.ts:172]: oooooh, no! Thank you - I already change it :D

**Pull request merged**